src/NSwag.CodeGeneration.TypeScript/Templates/Client.RequestUrl.liquid

```diff
@@ -6,9 +6,13 @@ if ({{ parameter.VariableName }} === undefined || {{ parameter.VariableName }} =
 {%    else -%}
 if ({{ parameter.VariableName }} !== null && {{ parameter.VariableName }} !== undefined)
 {%    endif -%}
-{%    if parameter.IsDateOrDateTimeArray -%}
+{%    if parameter.IsDateArray -%}
+url_ = url_.replace("{{ "{" }}{{ parameter.Name }}}", encodeURIComponent({{ parameter.VariableName }}.map(s_ => s_ ? formatDate(s_) : "null").join()));
+{%    elsif parameter.IsDateTimeArray -%}
 url_ = url_.replace("{{ "{" }}{{ parameter.Name }}}", encodeURIComponent({{ parameter.VariableName }}.map(s_ => s_ ? s_.{{ parameter.GetDateTimeToString }} : "null").join()));
-{%    elsif parameter.IsDateOrDateTime -%}
+{%    elsif parameter.IsDate -%}
+url_ = url_.replace("{{ "{" }}{{ parameter.Name }}}", encodeURIComponent({{ parameter.VariableName }} ? "" + formatDate({{ parameter.VariableName }}) : "null"));
+{%    elsif parameter.IsDateTime -%}
```

Comment:
Seems like this only half deals with the issue. Using this fix in a custom template does not generate the desired results (for me)

```ts
date(date: Date , cancelToken?: CancelToken | undefined): Promise<string> {
    let url_ = "/date?";
    if (date === undefined || date === null)
        throw new Error("The parameter 'date' must be defined and cannot be null.");
    else
        url_ += "date=" + encodeURIComponent(date ? "" + date.toISOString() : "") + "&";
    url_ = url_.replace(/[?&]$/, "");

    let options_: AxiosRequestConfig = {
        method: "GET",
        url: url_,
        headers: {
            "Accept": "application/json"
        },
        cancelToken
    };

    return this.instance.request(options_).catch((_error: any) => {
        if (isAxiosError(_error) && _error.response) {
            return _error.response;
        } else {
            throw _error;
        }
    }).then((_response: AxiosResponse) => {
        return this.processDate(_response);
    });
}
```

This is generated from a `DateOnly` ASP.NET Core API.

Seems like this should also deal with Query Parameters. Line 23 of `Client.RequestUrl.liquid`

```liquid
{% for parameter in operation.QueryParameters -%}
```

Changing the query parameters in addition to your change yields the desired output:

```ts
date(date: Date , cancelToken?: CancelToken | undefined): Promise<string> {
    let url_ = "/date?";
    if (date === undefined || date === null)
        throw new Error("The parameter 'date' must be defined and cannot be null.");
    else
        url_ += "date=" + encodeURIComponent(date ? "" + formatDate(date) : "") + "&";
    url_ = url_.replace(/[?&]$/, "");

    let options_: AxiosRequestConfig = {
        method: "GET",
        url: url_,
        headers: {
            "Accept": "application/json"
        },
        cancelToken
    };

    return this.instance.request(options_).catch((_error: any) => {
        if (isAxiosError(_error) && _error.response) {
            return _error.response;
        } else {
            throw _error;
        }
    }).then((_response: AxiosResponse) => {
        return this.processDate(_response);
    });
}
```

HOWEVER I don't understand where `formatDate` is coming from? It is undefined.